photon-api/src/test/scala/com/linkedin/photon/ml/data/LocalDataSetTest.scala

```diff
@@ -51,6 +53,254 @@ class LocalDataSetTest {
     }
   }
 
+  @DataProvider
+  def dataForRandomEffectFeatureSelection(): Array[Array[Any]] ={
+    val binaryIndices = Set[Int](0)
+    val binaryIndicesWithNonBinary = Set[Int](1)
+    val nonBinaryIndices = Set[Int]()
+    val nonBinaryIndicesWithNonBinary = Set[Int](0)
+    // First case x == 0 and y == 0
+    val x_0_y_0 = Array(
+      SparseVector(1.0),
+      SparseVector(1.0),
+      SparseVector(1.0),
+      SparseVector(1.0),
+      SparseVector(1.0)
+    )
+    val x_0_y_0_l = Array(0.0, 0.0, 0.0, 0.0, 0.0)
+    val x_0_y_0_expected = Map(0 -> 0.0)
+    val x_0_y_0_stats = BasicStatisticalSummary(
+      mean = Vector(0.5),
+      variance = Vector(0.0),
+      count = 50,
+      numNonzeros = Vector(25.0),
+      max = Vector(0.0),
+      min = Vector(0.0),
+      normL1 = Vector(0.0),
+      normL2 = Vector(0.0),
+      meanAbs = Vector(0.0),
+      None: Option[Int]
+    )
+    val x_0_y_0_g_positive = Array(0.0)
+
+    // Second case x == 0 and y != 0
```

Comment:
See my comments about splitting the responsibilities of the tests. The cases for `testComputeRatioCILowerBound` should be:

1. x = 0, y = 0
2. x = m, y = n
3. x = 0, 0 < y < n, epsilon = default
4. x = 0, 0 < y < n, epsilon = something else
5. x = m, 0 < y < n
6. 0 < x < m, 0 < y < n